homeassistant/components/enocean/binary_sensor.py

```diff
@@ -26,31 +36,90 @@
     }
 )
 
+YAML_BINARY_SENSORS_TO_IMPORT = "binary_sensors_to_import"
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Convert the config entry to an entity in the registry."""
+    if not hass.config_entries.async_entries(domain=DOMAIN):
+        return
+
+    config_entry = hass.config_entries.async_entries(domain=DOMAIN)[0]
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Binary Sensor platform for EnOcean."""
     dev_id = config.get(CONF_ID)
     dev_name = config.get(CONF_NAME)
     device_class = config.get(CONF_DEVICE_CLASS)
+    sensor = EnOceanBinarySensor(dev_id, dev_name, device_class)
+
+    entity_reg = await async_get_registry(hass)
+    configured_entities = async_entries_for_config_entry(
+        entity_reg, config_entry.entry_id
+    )
+
+    entities_to_migrate = hass.data[DATA_ENOCEAN].setdefault(
+        YAML_BINARY_SENSORS_TO_IMPORT, []
+    )
+    for entity in configured_entities:
+        if entity.unique_id == sensor.unique_id:
+            _LOGGER.warning(
+                "Yaml config for binary sensor %s ignored: entity exists in registry",
+                dev_name,
+            )
+            return
+
+    entities_to_migrate.append(EnOceanBinarySensor(dev_id, dev_name, device_class,))
```

Comment:
The entities should be created in `async_setup_entry`.

Reply:
Change done.